python/treelite/sklearn/exporter.py

```diff
@@ -0,0 +1,233 @@
+)
+
+
+def _export_tree(
+    model, *, tree_id, n_features, n_classes, n_targets, tree_depths, subestimator_class
+):
+    # pylint: disable=too-many-locals
+    try:
+        from sklearn import __version__ as sklearn_version
+        from sklearn.tree._tree import Tree as SKLearnTree
+    except ImportError as e:
+        raise TreeliteError("This function requires scikit-learn package") from e
+
+    tree_accessor = model.get_tree_accessor(tree_id)
+    has_categorical_split = tree_accessor.get_field("has_categorical_split").tolist()[0]
+    if has_categorical_split:
+        raise NotImplementedError(
+            "Trees with categorical splits cannot yet be exported as scikit-learn"
+        )
+
+    tree = SKLearnTree(n_features, n_classes, n_targets)
+
+    n_nodes = tree_accessor.get_field("num_nodes").tolist()[0]
+    nodes = np.empty(n_nodes, dtype=_node_dtype)
+
+    nodes["left_child"] = tree_accessor.get_field("cleft")
+    nodes["right_child"] = tree_accessor.get_field("cright")
+    nodes["feature"] = tree_accessor.get_field("split_index")
+    nodes["threshold"] = tree_accessor.get_field("threshold")
+    nodes["impurity"] = np.nan
+    nodes["n_node_samples"] = -1
+    nodes["weighted_n_node_samples"] = np.nan
+    nodes["missing_go_to_left"] = tree_accessor.get_field("default_left")
+
+    if n_targets == 1 and n_classes[0] == 1:
+        leaf_value = (
+            tree_accessor.get_field("leaf_value").astype("float64").reshape((-1, 1, 1))
+        )
+    else:
+        # Need to map leaf values to correct layout
+        leaf_value = np.zeros((n_nodes, n_targets, n_classes[0]), dtype="float64")
+        leaf_value_raw = tree_accessor.get_field("leaf_vector").astype("float64")
+        leaf_vec_begin = tree_accessor.get_field("leaf_vector_begin")
+        leaf_vec_end = tree_accessor.get_field("leaf_vector_end")
+        for node_id in range(n_nodes):
+            if leaf_vec_begin[node_id] != leaf_vec_end[node_id]:
+                # This node is a leaf node and outputs a vector
+                leaf_value[node_id, :, :] = leaf_value_raw[
+                    leaf_vec_begin[node_id] : leaf_vec_end[node_id]
+                ].reshape((n_targets, n_classes[0]))
+
+    state = {
+        "max_depth": tree_depths[tree_id],
+        "node_count": n_nodes,
+        "nodes": nodes,
+        "values": leaf_value,
+    }
+    tree.__setstate__(state)
+
+    subestimator = subestimator_class()
+    subestimator_state = {
+        "tree_": tree,
+        "n_outputs_": n_targets,
+        "_sklearn_version": sklearn_version,
+    }
+    if subestimator_class is DecisionTreeClassifier:
+        if n_targets == 1:
+            subestimator_state["n_classes_"] = n_classes[0]
+        else:
+            subestimator_state["n_classes_"] = n_classes.tolist()
+    subestimator.__setstate__(subestimator_state)
+    return subestimator
+
+
+def export_model(model: Model):
+    """
+    Export as scikit-learn RandomForest or GradientBoosting.
```

Comment:
```suggestion
    Export a model as a scikit-learn RandomForest.
```
clean up the gradient boosting bit that isn't supported.